### ISSUE TITLE
Joplin Client (Android and Linux AppImage) and WebDav Nextcloud Error

### ISSUE THREAD

**Issue by NDMann** [laurent22/joplin#3892]
## Environment

Joplin version: 1.2.6 (linux), 1.2.5 (android)
Platform: Linux, Android
OS specifics: Debian 10, Android 9

## Steps to reproduce

1. Run Android or Linux Appimage
2. Check Nextclud Admin Logs
3. [webdav] Fatal error appear

## Logfile

/tmp/.mount_JoplintW…evelopment.js:11494 Warning: componentWillReceiveProps has been renamed, and is not recommended for use. See https://fb.me/react-async-component-lifecycle-hooks for details.

* Move data fetching code or side effects to componentDidUpdate.
* If you're updating state whenever props change, refactor your code to use memoization techniques or move it to static getDerivedStateFromProps. Learn more at: https://fb.me/react-derived-state
* Rename componentWillReceiveProps to UNSAFE_componentWillReceiveProps to suppress this warning in non-strict mode. In React 17.x, only the UNSAFE_ name will work. To rename all deprecated lifecycles to their new names, you can run `npx react-codemod rename-unsafe-lifecycles` in your project source folder.

Please update the following components: Connect(MainScreenComponent), Connect(NavigatorComponent), Connect(NoteEditor), Connect(NoteListComponent), Connect(RootComponent), Connect(SearchBar), Connect(SideBarComponent), Provider

## Nextcloud logfile
[webdav] Fatal: Doctrine\DBAL\Exception\DeadlockException: An exception occurred while executing 'UPDATE `oc_filecache` SET `size` = GREATEST(?, `size` + ?) WHERE (`storage` = ?) AND (`path_hash` IN ('d41d8cd98f00b204e9800998ecf8427e', '45b963397aa40d4a0063e0d85e4fe7a1', '48a5be8ba95b084a287a36bf9d620b66', 'a2c02b09181e98c3539b43e163ea1f5f')) AND (`size` > '-1')' with params [-1, 83, 3]:

SQLSTATE[40001]: Serialization failure: 1213 Deadlock found when trying to get lock; try restarting transaction at <<closure>>

 0. /var/www/html/3rdparty/doctrine/dbal/lib/Doctrine/DBAL/DBALException.php line 169
    Doctrine\DBAL\Driver\AbstractMySQLDriver->convertException("An exception oc ... n", Doctrine\DBAL\Dr ... ]})
 1. /var/www/html/3rdparty/doctrine/dbal/lib/Doctrine/DBAL/DBALException.php line 149
    Doctrine\DBAL\DBALException::wrapException(Doctrine\DBAL\Driver\PDOMySql\Driver {}, Doctrine\DBAL\Dr ... ]}, "An exception oc ... n")
 2. /var/www/html/3rdparty/doctrine/dbal/lib/Doctrine/DBAL/Connection.php line 1071
    Doctrine\DBAL\DBALException::driverExceptionDuringQuery(Doctrine\DBAL\Driver\PDOMySql\Driver {}, Doctrine\DBAL\Dr ... ]}, "UPDATE `oc_file ... )", {1: -1,2: 83,3: 3})
 3. /var/www/html/lib/private/DB/Connection.php line 214
    Doctrine\DBAL\Connection->executeUpdate("UPDATE `oc_file ... )", [-1,83,3], [1,2,1])
 4. /var/www/html/3rdparty/doctrine/dbal/lib/Doctrine/DBAL/Query/QueryBuilder.php line 209
    OC\DB\Connection->executeUpdate("UPDATE `oc_file ... )", {dcValue1: -1,dcValue2: 83,dcValue3: 3}, {dcValue1: 1,dcValue2: 2,dcValue3: 1})
 5. /var/www/html/lib/private/DB/QueryBuilder/QueryBuilder.php line 217
    Doctrine\DBAL\Query\QueryBuilder->execute()
 6. /var/www/html/lib/private/Files/Cache/Propagator.php line 114
    OC\DB\QueryBuilder\QueryBuilder->execute()
 7. /var/www/html/lib/private/Files/Cache/HomePropagator.php line 49
    OC\Files\Cache\Propagator->propagateChange("*** sensitive parameter replaced ***", 1602330225, 83)
 8. /var/www/html/lib/private/Files/Cache/Updater.php line 140
    OC\Files\Cache\HomePropagator->propagateChange("*** sensitive parameter replaced ***", 1602330225, 83)
 9. /var/www/html/apps/dav/lib/Connector/Sabre/File.php line 310
    OC\Files\Cache\Updater->update("*** sensitive parameters replaced ***")
10. /var/www/html/apps/dav/lib/Connector/Sabre/Directory.php line 155
    OCA\DAV\Connector\Sabre\File->put(null)
11. /var/www/html/3rdparty/sabre/dav/lib/DAV/Server.php line 1104
    OCA\DAV\Connector\Sabre\Directory->createFile("sync_mobile_17c ... n", null)
12. /var/www/html/3rdparty/sabre/dav/lib/DAV/CorePlugin.php line 527
    Sabre\DAV\Server->createFile("files/user/Jo ... n", null, null)
13. /var/www/html/3rdparty/sabre/event/lib/WildcardEmitterTrait.php line 89
    Sabre\DAV\CorePlugin->httpPut(Sabre\HTTP\Request {}, Sabre\HTTP\Response {})
14. /var/www/html/3rdparty/sabre/dav/lib/DAV/Server.php line 474
    Sabre\DAV\Server->emit("method:PUT", [Sabre\HTTP\Requ ... }])
15. /var/www/html/3rdparty/sabre/dav/lib/DAV/Server.php line 251
    Sabre\DAV\Server->invokeMethod(Sabre\HTTP\Request {}, Sabre\HTTP\Response {})
16. /var/www/html/3rdparty/sabre/dav/lib/DAV/Server.php line 319
    Sabre\DAV\Server->start()
17. /var/www/html/apps/dav/lib/Server.php line 325
    Sabre\DAV\Server->exec()
18. /var/www/html/apps/dav/appinfo/v2/remote.php line 35
    OCA\DAV\Server->exec()
19. /var/www/html/remote.php line 167
    require_once("/var/www/html/a ... p")

PUT /remote.php/dav/files/user/Joplin/locks/sync_mobile_17cebc76ac934590b934ab144c53254e.json

**Comment by JazzXP** [comment]
Have you updated your NextCloud to the latest? I was having the same issue, and there was a known bug with WebDav in the version I was using.

**Comment by NDMann** [comment]
> Have you updated your NextCloud to the latest? I was having the same issue, and there was a known bug with WebDav in the version I was using.

Thanks @JazzXP
Yes I thought the same and I upgraded to Nextcloud 20.0.0 before opening this post.
I tried also to disable TOTP, but still same problem.
I'm anyway using App Password only for Joplin (as suggested from Joplin website).
What I have to say is: Joplin is anyway synchronizing. 
Only two things are the problem: 
- the alert in log is created too many times per day.
- the trash can is full of files named like this "sync_desktop_xxxxxxxxxxada,json"
Any ideas?
